src/wp-includes/block-supports/colors.php

```diff
@@ -16,13 +16,20 @@
  * @param WP_Block_Type $block_type Block Type.
  */
 function wp_register_colors_support( $block_type ) {
-	$color_support                 = property_exists( $block_type, 'supports' ) ? _wp_array_get( $block_type->supports, array( 'color' ), false ) : false;
-	$has_text_colors_support       = true === $color_support || ( is_array( $color_support ) && _wp_array_get( $color_support, array( 'text' ), true ) );
-	$has_background_colors_support = true === $color_support || ( is_array( $color_support ) && _wp_array_get( $color_support, array( 'background' ), true ) );
-	$has_gradients_support         = _wp_array_get( $color_support, array( 'gradients' ), false );
-	$has_link_colors_support       = _wp_array_get( $color_support, array( 'link' ), false );
-	$has_button_colors_support     = _wp_array_get( $color_support, array( 'button' ), false );
-	$has_heading_colors_support    = _wp_array_get( $color_support, array( 'heading' ), false );
+	$color_support = false;
+	if ( property_exists( $block_type, 'supports' ) ) {
```

Comment:
```suggestion
	if ( $block_type instanceof WP_Block_Type ) {
```

Reply:
Shouldn't this be part of a separate PR? I don't think it belongs here... 🤔